libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java

```diff
@@ -415,39 +375,18 @@ private void updateProgressChangeListener() {
   }
 
   private MapRouteProgressChangeListener buildMapRouteProgressChangeListener() {
-    shutdownVanishingRouteLineAnimator();
-    if (vanishRouteLineEnabled) {
-      vanishingRouteLineAnimator = new RouteLineValueAnimator();
-      vanishingRouteLineAnimator.setAnimationDelay(RouteConstants.ROUTE_LINE_VANISH_ANIMATION_DELAY);
-    } else {
-      vanishingRouteLineAnimator = null;
-    }
-
-    float vanishingPoint = (this.mapRouteProgressChangeListener != null)
-            ? this.mapRouteProgressChangeListener.getPercentDistanceTraveled() : 0f;
-    MapRouteProgressChangeListener newListener = new MapRouteProgressChangeListener(
-            this.routeLine,
-            routeArrow,
-            vanishingRouteLineAnimator
-    );
-    newListener.updatePercentDistanceTraveled(vanishingPoint);
-    return newListener;
+    return new MapRouteProgressChangeListener(this.routeLine, routeArrow);
   }
 
   /**
-   * Returns the percentage of the distance traveled that was last calculated. This is only
-   * calculated if the vanishing route line feature is enabled.
+   * Updates the appearance of the route line representing the part of the route that has
+   * been traveled.  The line appearance will be altered from the origin point to the point
+   * indicated by the @param point
    *
-   * @return the value calculated during the last progress update event or 0 if not enabled.
+   * @param point the end point of the traveled section of the route line.
    */
-  public float getPercentDistanceTraveled() {
-    return mapRouteProgressChangeListener.getPercentDistanceTraveled();
-  }
-
-  public void updateRouteLineWithDistanceTraveled(float distanceTraveled) {
-    routeLine.hideCasingLineAtOffset(distanceTraveled);
-    routeLine.hideRouteLineAtOffset(distanceTraveled);
-    mapRouteProgressChangeListener.updatePercentDistanceTraveled(distanceTraveled);
+  public void updateTraveledRouteLine(Point point) {
```

Comment:
Since the portion of the line behind the puck doesn't have to be transparent I went with the word "traveled". If a developer decides to make the color behind the puck a different color the line doesn't really vanish.

Reply:
That makes sense :+1: It'd be great to make it consistent and call the one in `MapRouteLine` the same way.

Reply:
Renamed the method in MapRouteLine.